keygen/src/main.c

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2020 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <jansson.h>
+#include <ctype.h>
+#include <errno.h>
+#include <openssl/ssl.h>
+#include <sodium.h>
+
+#include "auth.h"
+#include "iam.h"
+#include "ssl.h"
+#include "enftun/config.h"
+
+/**
+ * print_usage_msg() - Print usage message
+ * @name: The name of the program
+ *
+ * Prints usage message detailing parameters and usage
+ *
+ * Return: None
+ */
+static void print_usage_msg(char *name)
+{
+    const char *help_msg =
+        "Usage: %s -a NETWORK -c CONF_FILE -u API_EMAIL\n"
+        "NETWORK: The IPv6 subnet you are provisioning in. Eg. 2607:8F80:DEAD:BEEF::/64\n"
```

Comment:
A few suggestions for the help message.

We need to support both a network  (::/64) and a full address.  It would also be nice to provide a bit more explanation.

 So how about:
```
const char *help_msg =
    "Provision and register credentials (key and cert) for enftun to connect.\n"
    "\n"
    "Usage: %s -c CONF_FILE -a ADDRESS -u USERNAME\n"
    "CONF_FILE: Path to the config file for the enftun instance (/etc/enftun/enf0.conf)\n"
    "ADDRESS: The IPv6 address to provision credentials for. If a ::/64 network is provided, a new, randomly generated IPv6 address in the network will created.\n"
    "USERNAME: Your ENF API username.\n"
    "\n"
    "This program will prompt for your password.\n"
```